tests/components/cert_expiry/test_sensors.py

```diff
@@ -137,13 +140,11 @@ async def test_update_sensor(mock_now, hass):
 
     timestamp2 = future_timestamp(99)
 
-    next_update = dt_util.utcnow() + timedelta(hours=12)
-    async_fire_time_changed(hass, next_update)
-
     with patch(
         "homeassistant.components.cert_expiry.get_cert_expiry_timestamp",
         return_value=timestamp2,
     ):
+        await coordinator.async_refresh()
```

Comment:
Please see this test for how to patch and fire time correctly:
https://github.com/home-assistant/core/blob/f83d4e524b1f4bce5ef16a123ef8766304b1cdfb/tests/components/gdacs/test_geo_location.py#L43

Reply:
That method doesn't work here as-is since moving the clock forward affects the state of `cert_expiry` that's being tested:

1. The timestamp of the certificate is mocked to X days in the future (from the start of the test).
2. The `DataUpdateCoordinator` updates are tied to the clock and have large minimum intervals (12h).
3. The days-to-expiry sensor is also tied to the clock, as it compares the timestamp returned in (1) to the current time.

Firing time change events to move (2) forward aren't enough as the current time also needs to be patched differently at each update to handle (3) properly.

Reply:
I should have stuck to my gun to not use moving base time :/

Reply:
So then we need to change the return value of the time patch when moving forward.